src/components/motor_output_reordering/MotorOutputReorderingCanvas.js

```diff
@@ -0,0 +1,271 @@
+'use strict';
+
+class MotorOutputReorderCanvas
+{
+    constructor(canvas, droneConfiguration, motorClickCallback, spinMotorCallback)
+    {
+        this._spinMotorCallback = spinMotorCallback;
+        this._canvas = canvas;
+        this._motorClickCallback = motorClickCallback;
+        this._width = this._canvas.width();
+        this._height = this._canvas.height();
+        this._screenSize = Math.min(this._width, this._height);
+
+        this._config = new MotorOutputReorderConfig(this._screenSize);
+
+        // no component resize allowing yet
+        this._canvas.prop({
+            width: this._width,
+            height: this._height,
+        });
+
+        this._droneConfiguration = droneConfiguration;
+
+        this._ctx = this._canvas[0].getContext("2d");
+        this._ctx.translate(this._width / 2, this._height / 2);
+
+        this._canvas.mousemove((event) => {
+            this._onMouseMove(event);
+        });
+        this._canvas.mouseleave(() => {
+            this._onMouseLeave();
+        });
+        this._canvas.mousedown(() => {
+            this._onMouseDown();
+        });
+        this._canvas.mouseup(() => {
+            this._onMouseUp(event);
+        });
+        this._canvas.click(() => {
+            this._onMouseClick();
+        });
+
+        this.startOver();
+    }
+
+    pause()
+    {
+        this._keepDrawing = false;
+    }
+
+    startOver()
+    {
+        this.readyMotors = []; //motors that already being selected for remapping by user
+        this.remappingReady = false;
+        this._motorIndexToSpinOnMouseDown = -1;
+        this._keepDrawing = true;
+        this._mouse = {x : 0, y: 0};
+        window.requestAnimationFrame(() => {
+            this._drawOnce();
+        });
+    }
+
+    _drawOnce()
+    {
+        this._ctx.clearRect(- this._width / 2,  -this._height / 2, this._width, this._height);
```

Comment:
Here we have one parameter with space between `-` and `this` and the other without. I think is better without it.